pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/vector/lucene95/HnswVectorsFormat.java

```diff
@@ -19,10 +19,10 @@
 package org.apache.pinot.segment.local.segment.creator.impl.vector.lucene95;
 
 import java.io.IOException;
+import org.apache.lucene.backward_codecs.lucene95.Lucene95HnswVectorsFormat;
```

Comment:
@xiangfu0 / @Jackie-Jiang, this codec can not be used to write data since now it's in `backward_codec` package. unit test `LoaderTest.testVectorIndexLoad:766->constructSegmentWithVectorIndex:928` is failing with the following exception
```bash
Caused by: java.lang.UnsupportedOperationException: Old codecs may only be used for reading
	at org.apache.lucene.backward_codecs.lucene95.Lucene95HnswVectorsFormat.fieldsWriter(Lucene95HnswVectorsFormat.java:182)
```
based on the [conversation](https://lists.apache.org/thread/fl9ds5ofxgmq2obzh0j5hgsmp4n6jx65), I found, is it okay if I use lucene99 based codecs.

Reply:
Does this mean after upgrading Lucece to 0.9.10:
1. We can only write index with lucene99 based codecs?
2. If a new generated index is served by an old server (running Lucene 9.8.0), will it break?

Reply:
> Does this mean after upgrading Lucece to 0.9.10:
> 
> 1. We can only write index with lucene99 based codecs?

True, after upgrading Lucece to 0.9.10, we can only write an index based on lucene99.

> 2. If a new generated index is served by an old server (running Lucene 9.8.0), will it break?

I am unsure, but I will say it will break if I have to guess. The only thing is this vector functionality (classes) got added in Pinot on Dec/23, so I am not sure how much it is getting used. Maybe @xiangfu0 can help us here with a better understanding.